src/auth/ThreeStepOwned.sol

```diff
@@ -42,30 +43,37 @@ abstract contract ThreeStepOwned {
                              OWNERSHIP LOGIC
     //////////////////////////////////////////////////////////////*/
 
-    function renounceOwnership() public virtual onlyOwner {
-        owner = address(0);
-
-        emit OwnerUpdated(msg.sender, address(0));
-    }
-
-    function initOwnershipUpdate(address _newOwnerCandidate) public virtual onlyOwner {
-        _ownerCandidate = _newOwnerCandidate;
+    function setOwner(address newOwner) public virtual onlyOwner {
+        _ownerCandidate = newOwner;
 
-        emit OwnerUpdateInitiated(msg.sender, _newOwnerCandidate);
+        emit OwnerUpdateInitiated(msg.sender, newOwner);
     }
 
-    function confirmOwnershipUpdate() public virtual {
-        if (_updateConfirmedByCandidate) {
+    function confirmOwner() public virtual {
+        if (_ownerCandidateConfirmed) {
             require(msg.sender == owner, "UNAUTHORIZED");
-
-            _updateConfirmedByCandidate = false;
-            owner = _ownerCandidate;
-
-            emit OwnerUpdated(msg.sender, owner);
+            
+            delete _ownerCandidateConfirmed;
+            
+            address newOwner = _ownerCandidate;
+            
+            delete _ownerCandidate;
+            
+            owner = newOwner;
+
+            emit OwnerUpdated(msg.sender, newOwner);
         } else {
             require(msg.sender == _ownerCandidate, "UNAUTHORIZED");
 
-            _updateConfirmedByCandidate = true;
+            _ownerCandidateConfirmed = true;
         }
     }
+    
+    function renounceOwner() public virtual onlyOwner {
+        delete owner;
+        
+        delete _ownerCandidate;
```

Comment:
Does this worth it?

Reply:
Maybe not, actually. In that I guess there is no risk of the `_ownerCandidate` claiming `owner` if the `owner` is deleted.

Reply:
There isn't, I added a special test for that! 

As the current owner has always the last word (through the confirmation at the third step), if renounces is not authorized to perform that call.

Reply:
smart. So that works out for this ☑️